engine/src/nn/neuralnetapi.cpp

```diff
@@ -111,6 +111,8 @@ string NeuralNetAPI::get_device_name() const
 void NeuralNetAPI::validate_neural_network()
 {
     nnDesign.print();
+    
+    #ifndef MODE_STRATEGO
```

Comment:
Is it still necessary to use a `#ifndef` here?
The conditon has been relaxed and only displays warning messages instead of throwing an assert.

Reply:
Tested it, can be removed

Reply:
```suggestion
```